google-ads/src/main/java/com/google/ads/googleads/lib/GoogleAdsClient.java

```diff
@@ -149,16 +168,93 @@ public static Builder newBuilder() {
      */
     public abstract Builder setCredentials(Credentials credentials);
 
-    private void setCredentials(Properties properties) {
-      UserCredentials credentials =
-          UserCredentials.newBuilder()
-              .setClientId(properties.getProperty(ConfigPropertyKey.CLIENT_ID.getPropertyKey()))
-              .setClientSecret(
-                  properties.getProperty(ConfigPropertyKey.CLIENT_SECRET.getPropertyKey()))
-              .setRefreshToken(
-                  properties.getProperty(ConfigPropertyKey.REFRESH_TOKEN.getPropertyKey()))
-              .build();
-      setCredentials(credentials);
+    private void setCredentials(Properties properties) throws IOException {
```

Comment:
Since the IOException here is only for service account case, and we don't currently attempt to recover from this, what do you think about converting this into an unchecked exception?

Reply:
We already recover from it higher up in the stack, in `fromProperties()` on line 382. That's the only caller of this private method, so I think we're all set.